tensorflow/cc/gradients/math_grad.cc

```diff
@@ -687,6 +687,22 @@ Status MeanGrad(const Scope& scope, const Operation& op,
 }
 REGISTER_GRADIENT_OP("Mean", MeanGrad);
 
+Status ErfGrad(const Scope& scope, const Operation& op,
+               const std::vector<Output>& grad_inputs,
+               std::vector<Output>* grad_outputs) {
+  auto grad = grad_inputs[0];
+  auto two_over_root_pi = Cast(scope, Const(scope, 2 / std::sqrt(M_PI)),
+                               grad.type());
+  Scope grad_scope = scope.WithControlDependencies(grad);
+  auto x = ConjugateHelper(grad_scope, op.input(0));
+  auto dx = Mul(scope,
+                Mul(scope, grad, two_over_root_pi),
+                Exp(scope, Neg(scope, Square(scope, x))));
+  grad_outputs->push_back(dx);
+  return scope.status();
```

Comment:
After you use grad_scope for the scope usages after WithControlDependencies, you will need to `return grad_scope.status()`. It will automatically get the status of `scope` if that was an error.

Reply:
I think we should keep the WithControlDependencies so that we can match the behavior of the python gradients as best as possible. (Unless they are causing other issues.)